std/string.d

```diff
@@ -549,11 +565,19 @@ ptrdiff_t indexOf(T, size_t n)(ref T[n] s, in dchar c,
         If the parameters are not valid UTF, the result will still
         be in the range [-1 .. s.length], but will not be reliable otherwise.
   +/
-ptrdiff_t indexOf(Range)(Range s, in dchar c, in size_t startIdx,
+ptrdiff_t indexOf(Range)(Range sIn, in dchar c, in size_t startIdx,
         in CaseSensitive cs = CaseSensitive.yes)
-    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+    if ((isInputRange!Range && isSomeChar!(ElementEncodingType!Range)) ||
+        __traits(compiles, StringTypeOf!Range)
+    )
 {
-    static if (isSomeString!Range || (hasSlicing!Range && hasLength!Range))
+    static if (__traits(compiles, StringTypeOf!Range))
```

Comment:
I did
https://github.com/D-Programming-Language/phobos/pull/3651

people did not like it